### PR TITLE
Refactor(eos_cli_config_gen): Wildcard dict to list for ip-extended-community-lists-regexp

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`ip-extended-community-lists-regexp`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update all `host_vars` under molecule scenario `eos_cli_config_gen_v4.0` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Run molecule `cd ansible_collections/arista/avd/molecule ; make cli-4.0-schema`
  - [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Emil
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md [hunk 1]
```diff
@@ -669,7 +669,8 @@ ip_extcommunity_lists:
 
 ```yaml
 ip_extcommunity_lists_regexp:
-  < community_list_name >:
+  - name: < community_list_name >
+    types:
     - type: < permit | deny >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]: missing two extra spaces to be consistent with other models. This is also missing from the molecule file - causing the linter error in CI

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md [hunk 2]
```diff
@@ -669,7 +669,8 @@ ip_extcommunity_lists:
 
 ```yaml
 ip_extcommunity_lists_regexp:
-  < community_list_name >:
+  - name: < community_list_name >
+    types:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]: The secondary_key should be `entries`

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-extended-community-lists-regexp.j2 [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {# eos - ip-extended-community-lists-regexp #}
-{% for ip_extcommunity_list in ip_extcommunity_lists_regexp | arista.avd.natural_sort %}
+{% for ip_extcommunity_list in ip_extcommunity_lists_regexp | arista.avd.convert_dicts('name','types') | arista.avd.natural_sort('name') %}
 !
-{%     for sequence in ip_extcommunity_lists_regexp[ip_extcommunity_list] | arista.avd.default([]) %}
+{%     for sequence in ip_extcommunity_list.types | arista.avd.default([]) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-extended-community-lists-regexp.j2]: please change the item var to `entry` so `for entry in ip_extcommunity_list.entries` to make it more readable

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Thanks

**Review by emilarista — APPROVED** [review]
LGTM!

**Pull request merged**